lib/pages/chat_list/chat_list_view.dart

```diff
@@ -86,26 +86,33 @@ class ChatListView extends StatelessWidget {
       stream: Matrix.of(context).onShareContentChanged.stream,
       builder: (_, __) {
         final selectMode = controller.selectMode;
-        return WillPopScope(
-          onWillPop: () async {
+        return PopScope(
+          canPop: controller.selectMode == SelectMode.normal &&
+              !controller.isSearchMode &&
+              controller.activeFilter ==
+                  (AppConfig.separateChatTypes
+                      ? ActiveFilter.messages
+                      : ActiveFilter.allChats),
+          onPopInvoked: (pop) async {
+            if (pop) return;
             final selMode = controller.selectMode;
             if (controller.isSearchMode) {
               controller.cancelSearch();
-              return false;
+              return;
             }
             if (selMode != SelectMode.normal) {
```

Comment:
use `else if`s (applicable to multiple nearby lines)

Reply:
Fluffychat modified it in today's merge dev